src/gc.c

```diff
@@ -2540,6 +2563,7 @@ mark: {
             if (flags.how == 1) {
                 void *val_buf = jl_astaggedvalue((char*)a->data - a->offset * a->elsize);
                 verify_parent1("array", new_obj, &val_buf, "buffer ('loc' addr is meaningless)");
+                gc_heap_snapshot_record_internal_edge(new_obj, jl_valueof(val_buf));
```

Comment:
`val_buf` is not a jl_value_t. You can probably charge this size directly against `a` (e.g. with `gc_heap_snapshot_record_hidden_edge`, or maybe better yet, when computing the summarysize of `a` in `record_node_to_gc_snapshot`)